src/Ledger/Conway/Specification/Ratify.lagda.md

```diff
@@ -0,0 +1,719 @@
+
+  defaultDRepCredentialVotes : VDeleg ⇀ Vote
+  defaultDRepCredentialVotes = constMap (mapˢ vDelegCredential activeDReps) Vote.no
+
+  actualVotes : VDeleg ⇀ Vote
+  actualVotes  = castVotes ∪ˡ defaultDRepCredentialVotes
+                             ∪ˡ predeterminedDRepVotes
+
+  t : ℚ
+  t = maybe id 0ℚ (threshold (proj₁ pparams) (proj₂ <$> (proj₁ cc)) action DRep)
+
+  acceptedStake totalStake : Coin
+  acceptedStake  = ∑[ x ← stakeDistrVDeleg ∣ actualVotes ⁻¹ Vote.yes ] x
+  totalStake     = ∑[ x ← stakeDistrVDeleg ∣ dom (actualVotes ∣^ (❴ Vote.yes ❵ ∪ ❴ Vote.no ❵)) ] x
+```
+
+### Stake Pool Operator (<span class="AgdaInductiveConstructor">SPO</span>) Vote Counting {#sec:spo-vote-counting}
+
+This section defines the predicate `acceptedBySPO`{.AgdaFunction}, which uses the
+following auxiliary definitions.
+
++  `castVotes`{.AgdaFunction}: This map contains the votes that have been cast by
+   members of the `SPO`{.AgdaInductiveConstructor} body and have been collected as
+   part of the `GovActionState`{.AgdaDatatype} `gaSt`{.AgdaBound}.
+
++  `defaultVote`{.AgdaFunction}: This map sets a default vote to all
+   `SPOs`{.AgdaInductiveConstructor} who didn't vote, with the default depending on
+   the given action, and whether the `SPO`{.AgdaInductiveConstructor} has delegated
+   their vote to one of the default `DReps`{.AgdaInductiveConstructor}.
+
++  `actualVotes`{.AgdaFunction}: This map combines the votes cast by
+   `SPOs`{.AgdaInductiveConstructor} with `defaultVote`{.AgdaBound} using a
+   left-biased union making cast votes take precedence over default votes.
+
++  `t`{.AgdaFunction}: This rational number is the threshold used to calculate
+   whether the action is ratified by the `SPO`{.AgdaInductiveConstructor} body.
+
++  `acceptedStake`{.AgdaFunction}: This is the portion of
+   `SPO`{.AgdaInductiveConstructor} stake that voted
+   `yes`{.AgdaInductiveConstructor}; it is computed using the stake distribution
+   `stakeDistrPools`{.AgdaField} provided in the environment.
+
++  `totalStake`{.AgdaFunction}: This is the portion of
+   `SPO`{.AgdaInductiveConstructor} stake that voted either
+   `yes`{.AgdaInductiveConstructor} or `no`{.AgdaInductiveConstructor}; it is
+   computed using the stake distribution `stakeDistrPools`{.AgdaField} provided in
+   the environment.
+
+Let us discuss in more detail the way `SPO`{.AgdaInductiveConstructor} votes are
+counted, as the ledger specification's handling of this has evolved in response to
+community feedback.  Previously, if an `SPO`{.AgdaInductiveConstructor} did not vote,
+then it would be counted as having voted `abstain`{.AgdaInductiveConstructor} by
+default.  Members of the `SPO`{.AgdaInductiveConstructor} community found this
+behavior counterintuitive and requested that non-voters be assigned a
+`no`{.AgdaInductiveConstructor} vote by default, with the caveat that an
+`SPO`{.AgdaInductiveConstructor} could change its default setting by delegating its
```

Comment:
This community feedback discussion feels like it should go into some sort of admonition (since it is not essential to how ratification works)